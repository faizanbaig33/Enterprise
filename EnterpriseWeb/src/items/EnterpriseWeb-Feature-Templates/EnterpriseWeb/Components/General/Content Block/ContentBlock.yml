---
ID: "f0ff41eb-eb09-5af7-907a-279a823aecd4"
Parent: "eb1fbad8-007e-42c4-8d0d-3db80bc39417"
Template: "ab86861a-6030-46c5-b394-e8f99e8b87db"
Path: /sitecore/templates/Feature/EnterpriseWeb/Components/General/Content Block/ContentBlock
SharedFields:
- ID: "06d5295c-ed2f-4a54-9bf2-26228d113318"
  Hint: __Icon
  Value: Office/32x32/document_text.png
- ID: "12c33f3f-86c5-43a5-aeb4-5598cec45116"
  Hint: __Base template
  Value: |
    {B61A1D20-9EE5-4CCB-9E0D-24A97E58D293}
    {D47D8920-5527-4C26-92E8-BD999720B197}
    {3AA7D5FD-915F-4286-AA99-5020342711BD}
    {5FB8ACDF-E52A-4EE9-A3D9-83CD1C5C5CB1}
    {5E2F3CF0-45EC-4618-AA30-0465290A248D}
    {565E77B0-8234-479E-9F8F-0B44BDB36FC4}
    {E36771DC-C474-4722-B024-0305666A6A08}
    {206E2322-2899-408A-8C27-749A2C929D13}
- ID: "9c6106ea-7a5a-48e2-8cad-f0f693b1e2d4"
  Hint: __Read Only
  Value: 
- ID: "f7d48a55-2158-4f02-9356-756654404f73"
  Hint: __Standard values
  Value: "{087D0CC2-2D4A-5536-A111-2886A97538CF}"
Languages:
- Language: en
  Fields:
  - ID: "b5e02ad9-d56f-4c41-a065-a133db87bdeb"
    Hint: __Display name
    Value: Content Block
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20210707T230223Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "828a79c2-886b-4918-8521-53d12cac586c"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230224T045354Z
