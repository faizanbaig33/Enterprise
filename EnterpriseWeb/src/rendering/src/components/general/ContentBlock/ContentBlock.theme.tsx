import classNames from 'classnames';
import { ThemeFile } from 'lib/context/ThemeContext';
import { ComponentAlignment, BackgroundColor } from './ContentBlock';

const getDynamicButtonStyles = (backgroundColor: BackgroundColor): string => {
  return classNames(`${backgroundColor === 'black' ? 'text-primary' : ''}`);
};

const getDynamicStyles = (alignment: ComponentAlignment): string => {
  return classNames(
    `${
      alignment === 'left' ? 'col-span-12' : 'col-span-12 col-start-0 md:col-span-10 md:col-start-2'
    }`
  );
};

export const ContentBlockTheme = (
  alignment: ComponentAlignment,
  backgroundColor: BackgroundColor
): ThemeFile => {
  return {
    aw: {
      classes: {
        contentWrapper: classNames(
          backgroundColor != 'white' ? 'py-l' : '',
          getDynamicStyles(alignment)
        ),
        headlineClass: 'text-theme-text text-sm-m md:text-m font-bold mb-s',
        bodyClass: 'text-theme-body text-body mb-s',
        buttonGroupClass: {
          wrapper: 'flex-col',
          cta1Classes: 'mr-2 mb-m md:mb-0',
          cta2Classes: 'ml-xs md:ml-0 px-s md:px-0',
        },
      },
    },
    rba: {
      classes: {
        contentWrapper: classNames(
          backgroundColor != 'white' ? 'py-l' : '',
          getDynamicStyles(alignment)
        ),
        headlineClass: 'text-theme-text text-sm-m md:text-m font-medium mb-s',
        bodyClass: 'text-theme-body text-body mb-s',
        buttonGroupClass: {
          wrapper: 'flex-col md:items-center',
          cta1Classes: 'mr-2',
          cta2Classes: classNames(getDynamicButtonStyles(backgroundColor), 'my-s md:my-0 mx-s'),
        },
      },
    },
  };
};
