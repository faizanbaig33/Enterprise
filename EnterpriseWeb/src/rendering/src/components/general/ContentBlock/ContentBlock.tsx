// Components
import { BodyCopy } from 'src/helpers/BodyCopy';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { Component } from 'src/helpers/Component';
import { ContentBlockTheme } from './ContentBlock.theme';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { getEnum } from 'lib/utils';
import { Headline } from 'src/helpers/Headline';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';

export type ContentBlockProps = Feature.EnterpriseWeb.Components.General.ContentBlock.ContentBlock;

export type ComponentAlignment = 'left' | 'center';
export type BackgroundColor = 'black' | 'gray' | 'white';

const ContentBlock = (props: ContentBlockProps): JSX.Element => {
  const alignment = getEnum<ComponentAlignment>(props.fields?.alignment) || 'left';
  const backgroundColor = getEnum<BackgroundColor>(props.fields?.backgroundColor) || 'white';
  const { themeData } = useTheme(ContentBlockTheme(alignment, backgroundColor));

  return (
    <Component
      variant="lg"
      backgroundVariant={getEnum(props.fields?.backgroundColor)}
      dataComponent="general/contentblock"
      {...props}
    >
      <div className={themeData.classes.contentWrapper}>
        <Headline defaultTag="h2" classes={themeData.classes.headlineClass} {...props} />
        <BodyCopy classes={themeData.classes.bodyClass} {...props} />
        <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<ContentBlockProps>(ContentBlock);
