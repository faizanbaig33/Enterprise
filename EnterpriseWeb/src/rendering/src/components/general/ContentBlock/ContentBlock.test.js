// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import ContentBlock from './ContentBlock';
import defaultData from './ContentBlock.mock-data-center';

it('renders correctly', () => {
  const component = snapshot(ContentBlock, { componentProps: defaultData });
  hasDataComponent(component, 'general/contentblock');
});
