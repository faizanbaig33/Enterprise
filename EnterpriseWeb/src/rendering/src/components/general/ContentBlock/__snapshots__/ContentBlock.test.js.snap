// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/contentblock"
  id="content block left"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="py-l col-span-12 col-start-0 md:col-span-10 md:col-start-2"
      >
        <div
          class="items-top font-sans  text-theme-text text-sm-m md:text-m font-bold mb-s"
        >
          <h2>
            This is a headline lorem ipsum dolor amit
          </h2>
        </div>
        <div
          class="text-theme-body text-body mb-s"
        >
          <div
            class="body-copy"
          >
            <div>
              Purus sit amet volutpat consequat mauris nunc congue nisi. Eget arcu dictum varius duis at consectetur. Mi proin sed libero enim sed faucibus turpis in. Feugiat vivamus at augue eget arcu dictum varius. Urna cursus eget nunc scelerisque viverra mauris in aliquam. Sem integer vitae justo eget magna fermentum.

              <br />
              

              <br />
              
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Enim sed faucibus turpis in eu mi bibendum neque egestas. Pharetra diam sit amet nisl suscipit adipiscing. Mi tempus imperdiet nulla malesuada pellentesque. Duis at consectetur lorem donec massa sapien faucibus et molestie. Risus quis varius quam quisque id diam vel quam elementum.
            </div>
          </div>
        </div>
        <div
          class="flex-col mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray mr-2 mb-m md:mb-0 mr-m"
            href="/"
            role="link"
            tabindex="0"
            title=""
          >
            Call to Action
            <span
              class="ml-xxs"
            />
          </a>
          <a
            class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray ml-xs md:ml-0 px-s md:px-0"
            href="/"
            role="link"
            tabindex="0"
            title=""
          >
            Call to Action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
